.buildkite/commands/build-and-test.sh

```diff
@@ -7,5 +7,24 @@ gem install bundler
 echo "--- :rubygems: Setting up Gems"
 install_gems
 
-echo "--- :hammer_and_wrench: Building and testing"
-bundle exec fastlane test
+echo "--- Verify BUILDKITE_ANALYTICS_TOKEN"
+if [ -z "$BUILDKITE_ANALYTICS_TOKEN" ]; then
+  # This should not run because of the `-u` in the shebang, but just in case...
+  echo "+++ Token not available!"
+  exit 1
+else
+  echo "+++ Test Analytics token found, moving on"
+fi
+
+echo "--- :hammer_and_wrench: Building for testing"
+bundle exec fastlane build_for_testing
+
+echo "--- Hack to pass token to the xctestrun"
+# TODO: In real life, the env var name, value, and xctestrun path should be
+# parameters read or injected from somewhere
+/usr/libexec/PlistBuddy -c \
+  "add TestPressTests:EnvironmentVariables:BUILDKITE_ANALYTICS_TOKEN string $BUILDKITE_ANALYTICS_TOKEN" \
+  ./DerivedData/Build/Products/TestPress_iphonesimulator15.5-x86_64.xctestrun
+
+echo "--- :microscope: Running unit tests"
+bundle exec fastlane test_without_building
```

Comment:
Here's the three-steps process to give the unit tests access to the `BUILDKITE_ANALYTICS_TOKEN` so they can report the results to Test Analytics

1. Run `xcodebuild build-for-testing` to build an `xctestrun` file
2. Use `PlistBuddy` to add `BUILDKITE_ANALYTICS_TOKEN` to the `xctestrun` environment
3. Run the tests from the `xctestrun` with `xcodebuild test-without-building`

The code above uses Fastlane because that's what we use in WordPress and WooCommerce, but Fastlane itself uses `xcodebuild` under the hood, so the process _should_ work 1-1 with vanilla `xcodebuild`, although I haven't tried it.